hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/scope/HandleHederaOperationsTest.java

```diff
@@ -249,16 +262,51 @@ void createContractWithParentDispatchesAsExpectedThenMarksCreated() {
 
         subject.createContract(666L, NON_SYSTEM_ACCOUNT_ID.accountNumOrThrow(), CANONICAL_ALIAS);
 
-        verify(context)
-                .dispatchRemovableChildTransaction(
-                        eq(synthTxn),
-                        eq(ContractCreateRecordBuilder.class),
-                        any(Predicate.class),
-                        eq(A_NEW_ACCOUNT_ID));
+        assertInternalFinisherAsExpected(captor.getValue(), synthContractCreation);
         verify(tokenServiceApi)
                 .markAsContract(AccountID.newBuilder().accountNum(666L).build(), NON_SYSTEM_ACCOUNT_ID);
     }
 
+    private void assertInternalFinisherAsExpected(
+            @NonNull final UnaryOperator<Transaction> internalFinisher,
+            @NonNull final ContractCreateTransactionBody expectedOp)
+            throws IOException {
+        Objects.requireNonNull(internalFinisher);
+
+        // The finisher should swap the crypto create body with the contract create body
+        final var cryptoCreateBody = TransactionBody.newBuilder()
+                .cryptoCreateAccount(CryptoCreateTransactionBody.DEFAULT)
+                .build();
+        final var cryptoCreateInput = Transaction.newBuilder()
+                .signedTransactionBytes(SignedTransaction.PROTOBUF.toBytes(SignedTransaction.newBuilder()
+                        .bodyBytes(TransactionBody.PROTOBUF.toBytes(cryptoCreateBody))
+                        .build()))
+                .build();
+        final var cryptoCreateOutput = internalFinisher.apply(cryptoCreateInput);
+        final var finishedBody = TransactionBody.PROTOBUF.parseStrict(SignedTransaction.PROTOBUF
+                .parseStrict(cryptoCreateOutput.signedTransactionBytes().toReadableSequentialData())
+                .bodyBytes()
+                .toReadableSequentialData());
+        assertEquals(expectedOp, finishedBody.contractCreateInstanceOrThrow());
```

Comment:
Just a comment: This kind of test is a place you could use soft assertions.  You have three assertions here in one custom assert method - but though they're related they're not dependent (I don't think).  And so with soft assertions you could , in one test, get _all_ failures of the finisher, not just the first to hit.  For example.

Reply:
Ah yes stillll need to learn `AssertJ` 😳 .